dll/win32/shell32/folders/CDesktopFolder.h

```diff
@@ -23,6 +23,73 @@
 #ifndef _CDESKTOPFOLDER_H_
 #define _CDESKTOPFOLDER_H_
 
+class CStubFolderBase : public IShellFolder
+{
+public:
+    CStubFolderBase() { }
+
+    STDMETHODIMP QueryInterface(REFIID riid, void **ppvObj) override { return E_NOTIMPL; }
+    STDMETHODIMP_(ULONG) AddRef()  override { return 3; }
+    STDMETHODIMP_(ULONG) Release() override { return 2; }
+
+    // IShellFolder methods
+    STDMETHODIMP ParseDisplayName(HWND hwndOwner, LPBC pbc, LPOLESTR lpszDisplayName,
+                                  DWORD *pchEaten, PIDLIST_RELATIVE *ppidl, DWORD *pdwAttributes) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP EnumObjects(HWND hwndOwner, DWORD dwFlags, LPENUMIDLIST *ppEnumIDList) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP BindToObject(PCUIDLIST_RELATIVE pidl, LPBC pbcReserved, REFIID riid, LPVOID *ppvOut) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP BindToStorage(PCUIDLIST_RELATIVE pidl, LPBC pbcReserved, REFIID riid, LPVOID *ppvOut) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl1, PCUIDLIST_RELATIVE pidl2) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP CreateViewObject(HWND hwndOwner, REFIID riid, LPVOID *ppvOut) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP GetAttributesOf(UINT cidl, PCUITEMID_CHILD_ARRAY apidl, DWORD *rgfInOut) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP GetUIObjectOf(HWND hwndOwner, UINT cidl, PCUITEMID_CHILD_ARRAY apidl,
+                               REFIID riid, UINT * prgfInOut, LPVOID * ppvOut) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP GetDisplayNameOf(PCUITEMID_CHILD pidl, DWORD dwFlags, LPSTRRET strRet) override
+    { return E_NOTIMPL; }
+    STDMETHODIMP SetNameOf(HWND hwndOwner, PCUITEMID_CHILD pidl, LPCOLESTR lpName,
+                           DWORD dwFlags, PITEMID_CHILD *pPidlOut) override
+    { return E_NOTIMPL; }
+};
+
+class CShellUrlStub : public CStubFolderBase
+{
+public:
+    STDMETHODIMP
+    ParseDisplayName(HWND hwndOwner, LPBC pbc, LPOLESTR lpszDisplayName, DWORD *pchEaten,
```

Comment:
Why are this objects?
Are they ever used as objects / do they need something else other than a `ParseDisplayName` ?

Reply:
When `URL_SCHEME_MSSHELLROOTED`, Windows does something complicated. It needs an object.

Reply:
That does not seem to be reflected in this PR?
There are a bunch of objects created just to call a single function.

Reply:
No, that doesn't. The case of `URL_SCHEME_MSSHELLROOTED` was not implemented.